commit a8125ac8216ef26071dbd552cd48e6c97de90ef6
Author: Yuriy Denyshchenko <ydenyshchenk@magento.com>
Date:   Tue Jan 24 16:42:01 2017 +0200

    MDVA-2600: Email logo not appearing

diff --git a/vendor/magento/module-config/Model/Config/Backend/Email/Logo.php b/vendor/magento/module-config/Model/Config/Backend/Email/Logo.php
index 701439ff2db..b88923a6173 100644
--- a/vendor/magento/module-config/Model/Config/Backend/Email/Logo.php
+++ b/vendor/magento/module-config/Model/Config/Backend/Email/Logo.php
@@ -11,6 +11,9 @@
  */
 namespace Magento\Config\Model\Config\Backend\Email;
 
+/**
+ * @deprecated
+ */
 class Logo extends \Magento\Config\Model\Config\Backend\Image
 {
     /**
diff --git a/vendor/magento/module-email/Model/AbstractTemplate.php b/vendor/magento/module-email/Model/AbstractTemplate.php
index eb9ed2ead8f..4ce6f0d90a6 100644
--- a/vendor/magento/module-email/Model/AbstractTemplate.php
+++ b/vendor/magento/module-email/Model/AbstractTemplate.php
@@ -389,7 +389,7 @@ abstract class AbstractTemplate extends AbstractModel implements TemplateTypesIn
             $store
         );
         if ($fileName) {
-            $uploadDir = \Magento\Config\Model\Config\Backend\Email\Logo::UPLOAD_DIR;
+            $uploadDir = \Magento\Email\Model\Design\Backend\Logo::UPLOAD_DIR;
             $mediaDirectory = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA);
             if ($mediaDirectory->isFile($uploadDir . '/' . $fileName)) {
                 return $this->storeManager->getStore()->getBaseUrl(
diff --git a/vendor/magento/module-email/Model/Design/Backend/Logo.php b/vendor/magento/module-email/Model/Design/Backend/Logo.php
new file mode 100644
index 00000000000..f315af490d2
--- /dev/null
+++ b/vendor/magento/module-email/Model/Design/Backend/Logo.php
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Email\Model\Design\Backend;
+
+use Magento\Theme\Model\Design\Backend\Logo as DesignLogo;
+
+class Logo extends DesignLogo
+{
+    /**
+     * The tail part of directory path for uploading
+     */
+    const UPLOAD_DIR = 'email/logo';
+
+    /**
+     * Upload max file size in kilobytes
+     *
+     * @var int
+     */
+    protected $maxFileSize = 2048;
+
+    /**
+     * Getter for allowed extensions of uploaded files
+     *
+     * @return string[]
+     */
+    public function getAllowedExtensions()
+    {
+        return ['jpg', 'jpeg', 'gif', 'png'];
+    }
+}
diff --git a/vendor/magento/module-email/etc/di.xml b/vendor/magento/module-email/etc/di.xml
index 5be5efeb0b6..d1477d470ab 100644
--- a/vendor/magento/module-email/etc/di.xml
+++ b/vendor/magento/module-email/etc/di.xml
@@ -27,7 +27,7 @@
                 <item name="email_logo" xsi:type="array">
                     <item name="path" xsi:type="string">design/email/logo</item>
                     <item name="fieldset" xsi:type="string">other_settings/email</item>
-                    <item name="backend_model" xsi:type="string">Magento\Theme\Model\Design\Backend\Logo</item>
+                    <item name="backend_model" xsi:type="string">Magento\Email\Model\Design\Backend\Logo</item>
                     <item name="base_url" xsi:type="array">
                         <item name="type" xsi:type="string">media</item>
                         <item name="scope_info" xsi:type="string">1</item>
diff --git a/vendor/magento/module-theme/Model/Design/Backend/Logo.php b/vendor/magento/module-theme/Model/Design/Backend/Logo.php
index 78bc7603492..05b4bb9e272 100644
--- a/vendor/magento/module-theme/Model/Design/Backend/Logo.php
+++ b/vendor/magento/module-theme/Model/Design/Backend/Logo.php
@@ -21,7 +21,7 @@ class Logo extends Image
      */
     protected function _getUploadDir()
     {
-        return $this->_mediaDirectory->getRelativePath($this->_appendScopeInfo(self::UPLOAD_DIR));
+        return $this->_mediaDirectory->getRelativePath($this->_appendScopeInfo(static::UPLOAD_DIR));
     }
 
     /**
